appinventor/blocklyeditor/src/blocks/procedures.js

```diff
@@ -902,3 +902,49 @@ Blockly.Blocks['procedures_callreturn'] = {
   typeblock: [{ translatedName: Blockly.Msg.LANG_PROCEDURES_CALLRETURN_TRANSLATED_NAME}]
 };
 
+Blockly.Blocks['map_list2'] = {
+  // Map a list with a specific procedure
+  category: 'Procedures',
+  helpUrl: "helpUrl",
+  init: function () {
+    this.setColour(Blockly.PROCEDURE_CATEGORY_HUE);
+    var procDb = this.getTopWorkspace().getProcedureDatabase();
+    var procNameArray = [Blockly.FieldProcedure.defaultValue];
```

Comment:
I think the reason you're seeing duplicate values is because you create procNameArray outside of the closure, so it will remember everything added on each call to procNamesFxn. Move this variable declaration inside of the procNamesFxn so that it will be freshly created on each call.

Reply:
I see. Putting procNameArray inside procNamesFxn did fix the issue. Thank you very much for your help!